examples/interactive_bluefog_helloworld.ipynb

```diff
@@ -0,0 +1,634 @@
+    "```\n",
+    "where `machine1` to `machine4` are the machine name or the address of the machine and `:4` after the machine name means on that machine launch 4 processes. Before you run this command, please make sure that all the machines is able to be login through `ssh` **without** typing the password.\n",
+    "\n",
+    "That's it!\n",
+    "\n",
+    "**Other remakrs**\n",
+    "- Please make sure the starting jupyter notebook command and the ibfrun command are under the same machine.\n",
+    "- If you want to communicate over GPUs through NCCL, the number of process in each machine should not exceed the number GPUs instead of CPUs.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-01-02T01:57:44.573421Z",
+     "start_time": "2021-01-02T01:57:43.495531Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import torch\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Connect the Jupyter Notebook to the Cluster (workers)\n",
+    "\n",
+    "The first step is to let the jupyter notebook connect to cluster. To achieve that, import the ipypyarallel library since we rely on it to manager the cluster, then connect it through `ipp.Client(profile=\"bluefog\")` as shown in the following cell.\n",
+    "\n",
+    "`rc` (remote client) is the key that you will use to send the code from this notebook (driver) to the cluster namely multiple workers. \n",
+    "To check the notebook has connected to the worker succesfully, you can look the ID of remote clients through `rc.ids`\n",
+    "Depending on how many processes you starts, the length of rc will be the same number as the number of process you start the `ibfrun`, i.e. the number after `-np`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-01-02T01:57:45.309973Z",
+     "start_time": "2021-01-02T01:57:45.285024Z"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[0, 1, 2, 3]"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import ipyparallel as ipp\n",
+    "rc = ipp.Client(profile=\"bluefog\")\n",
+    "rc.ids"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you saw the `ids` is an empty list, this means ipyparallel cannot connect it with the clusters. You can stop the `ibfrun` by `ctrl+c`. Kill all remainning `ipcontroller` or `mpi` processes through `ps` if necessary. Then, try to rerun the `ibfrun` again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#  Parallel execution through `px`\n",
```

Comment:
Consider add an example for %px somewhere.

Reply:
Added